annlite/container.py

```diff
@@ -136,6 +136,48 @@ def ivf_search(
             doc_ids.append(doc_id)
         return dists, doc_ids, cell_ids
 
+    def filter_cells(
+        self,
+        cells: 'np.ndarray',
+        where_clause: str = '',
+        where_params: Tuple = (),
+        limit: int = 10,
+        include_metadata: bool = False,
+    ):
+
+        filtered_docs_list = []
+        for cell_id in cells:
+            cell_table = self.cell_table(cell_id)
+            cell_size = cell_table.count()
+            if cell_size == 0:
+                continue
+
+            indices = None
+            if where_clause or (cell_table.deleted_count() > 0):
+                indices = cell_table.query(
+                    where_clause=where_clause, where_params=where_params
+                )
+
+                if len(indices) == 0:
+                    continue
+
+                indices = np.array(indices, dtype=np.int64)
+
+            filtered_cell_docs = DocumentArray()
+            for doc_id in indices:
```

Comment:
the results from `cell_table.query()` is the offset integer, rather than the str doc id. We need get the doc id accordingly:
```
doc_id = self.cell_table(cell_id).get_docid_by_offset(offset)
```